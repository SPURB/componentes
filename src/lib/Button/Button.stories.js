import { storiesOf } from '@storybook/vue'
import Button from './Button.vue'

storiesOf('Button', module).add('Button', () => ({
  components: { Button },
  template: `<Button action-text="submit" @buttonClick="getClick" />`,
  methods: {
    getClick () {
      alert('Bot√£o clicado!')
    }
  }
}))
