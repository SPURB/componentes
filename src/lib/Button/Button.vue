<template>
  <button 
    :class="style"
    class="text-white font-bold py-2 px-4 rounded"
    @click="actionButton"
  >
    {{ actionText }}
  </button>
</template>

<script>
export default {
  name: 'Button',
  props: {
    actionText: {
      type: String,
      required: true
    },
    styleButton: {
      type: String,
      required: true,
      default: 'bg-blue-500 hover:bg-blue-700'
    }
  },
  methods: {
    actionButton () {
      this.$emit('buttonClick')
    }
  }
}
</script>

<style>

</style>