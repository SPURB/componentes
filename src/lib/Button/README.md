# Button.vue
Componente de botão

## Exemplo
https://spurb.github.io/componentes/?path=/story/button--button

## Instruções de uso
Declare
```js
import { Button } from '@spurb/componentes'
export default {
  components: { Button }
}
```

Exemplo de uso no template
```html
<Button action-text="submit" @buttonClick="getClick" />
```

### Props
| nome | data type | descrição |
|:-|:-|:-|
| `actionText` |  String | Define o texto do botão |
| `styleButton` |  String | Define o estilo do botão |

### Event
| nome | descrição |
|:-|:-|
| `@buttonClick` | Envia um action quando o botão for clicado |

