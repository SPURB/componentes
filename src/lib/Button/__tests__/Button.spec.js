import { shallowMount } from '@vue/test-utils'
import Button from '../Button.vue'

const actionText = 'submit'
const styleButton = 'bg-blue-500 hover:bg-blue-700'

describe('Button.vue', () => {
  it('Renderiza corretamente passando props', () => {
    const wrapper = shallowMount(Button, {
      propsData: { actionText, styleButton }
    })
    expect(wrapper.element).toMatchSnapshot() // compara renderização por snashot
  })
})
