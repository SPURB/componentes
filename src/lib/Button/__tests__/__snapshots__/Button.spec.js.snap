// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.vue Renderiza corretamente passando props 1`] = `
<button
  class="text-white font-bold py-2 px-4 rounded"
>
  
  submit

</button>
`;
